src/core/service.c

```diff
@@ -2144,6 +2145,54 @@ _pure_ static bool service_can_reload(Unit *u) {
         return !!s->exec_command[SERVICE_EXEC_RELOAD];
 }
 
+static int service_exec_command_index(Unit *u, ServiceExecCommand id, bool control) {
+        Service *s = SERVICE(u);
+        int index = 0;
+        ExecCommand *current, *first, *c;
+
+        assert(s);
+
+        first = s->exec_command[id];
+        current = control ? s->control_command : s->main_command;
+
+        /* Figure out where we are in the list be walking back to the beginning */
+        for (c = current; c != first; c = c->command_prev)
+                index++;
+
+        return index;
+}
```

Comment:
given that this function cannot really fail, and can't validly return a negative number I think it would be best to use an unsigned return type for this, such as simply "unsigned"